src/planner/vectorfield/VectorFieldPlanner.cpp

```diff
@@ -1,278 +1,204 @@
 
 //==============================================================================
-static void checkCollision(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& stateSpace,
-    const aikido::constraint::TestablePtr& constraint)
+std::unique_ptr<aikido::trajectory::Spline> followVectorField(
+    const aikido::planner::vectorfield::VectorField* vectorField,
+    const aikido::statespace::StateSpace::State* startState,
+    const aikido::constraint::Testable* constraint,
+    std::chrono::duration<double> timelimit,
+    double initialStepSize,
+    double checkConstraintResolution,
+    planner::PlanningResult* planningResult)
 {
-  // Get current position
-  auto state = stateSpace->getScopedStateFromMetaSkeleton();
-  // Throw a termination if in collision
-  if (!constraint->isSatisfied(state))
-  {
-    throw VectorFieldTerminated("state in collision");
-  }
-}
+  using namespace std::placeholders;
+  using errorStepper = boost::numeric::odeint::
+      runge_kutta_dopri5<Eigen::VectorXd,
+                         double,
+                         Eigen::VectorXd,
+                         double,
+                         boost::numeric::odeint::vector_space_algebra>;
 
-//==============================================================================
-std::unique_ptr<aikido::trajectory::Spline> planPathByVectorField(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    const aikido::constraint::TestablePtr& _constraint,
-    double _dt,
-    const VectorFieldCallback& _vectorFiledCb,
-    const VectorFieldStatusCallback& _statusCb)
-{
-  auto saver = MetaSkeletonStateSpaceSaver(_stateSpace);
-  DART_UNUSED(saver);
+  std::size_t dimension = vectorField->getStateSpace()->getDimension();
+  auto integrator = std::make_shared<detail::VectorFieldIntegrator>(
+      vectorField, constraint, timelimit.count(), checkConstraintResolution);
 
-  dart::dynamics::MetaSkeletonPtr skeleton = _stateSpace->getMetaSkeleton();
+  integrator->start();
 
-  if (skeleton->getPositionUpperLimits() == skeleton->getPositionLowerLimits())
-  {
-    throw std::invalid_argument("State space volume zero");
-  }
-  if (skeleton->getVelocityUpperLimits() == skeleton->getVelocityLowerLimits())
+  try
   {
-    throw std::invalid_argument("Velocity space volume zero");
+    Eigen::VectorXd initialQ(dimension);
+    vectorField->getStateSpace()->logMap(startState, initialQ);
+
+    // Integrate the vector field to get a configuration space path.
+    boost::numeric::odeint::integrate_adaptive(
+        errorStepper(),
+        std::bind(&detail::VectorFieldIntegrator::step, integrator, _1, _2, _3),
+        initialQ,
+        0.,
+        integrationTimeInterval,
+        initialStepSize,
+        std::bind(&detail::VectorFieldIntegrator::check, integrator, _1, _2));
   }
-
-  for (std::size_t i = 0; i < skeleton->getNumDofs(); ++i)
+  catch (const detail::VectorFieldTerminated& e)
```

Comment:
Add a comment explaining that `VectorFieldTerminated` is an exception that we raise internally to terminate integration, i.e. it does not indicate that an error has actually occurred.